lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -451,9 +458,14 @@ private ColumnChunk readPrimitive(PrimitiveField field)
         return columnChunk;
     }
 
-    public Map<String, Metric<?>> getCodecMetrics()
+    public Metrics getMetrics()
     {
-        return codecMetrics;
+        if (columnIndexRowsFiltered < 0) {
+            return new Metrics(codecMetrics);
+        }
+        ImmutableMap.Builder<String, Metric<?>> metrics = ImmutableMap.<String, Metric<?>>builder()
+                .putAll(codecMetrics);
+        return new Metrics(metrics.put(COLUMN_INDEX_ROWS_FILTERED, new LongCount(columnIndexRowsFiltered)).buildOrThrow());
```

Comment:
This would be easier to read as:
```suggestion
        ImmutableMap.Builder<String, Metric<?>> metrics = ImmutableMap.<String, Metric<?>>builder()
                .putAll(codecMetrics);

        if (columnIndexRowsFiltered >= 0) {
            metrics.put(COLUMN_INDEX_ROWS_FILTERED, new LongCount(columnIndexRowsFiltered))
        }
        
        return new Metrics(metrics.buildOrThrow());
```

Reply:
I wanted to avoid using the map builder when we're not exporting COLUMN_INDEX_ROWS_FILTERED. Since `codecMetrics` is an immutable map itself, we can return that directly for `columnIndexRowsFiltered < 0` case.

Reply:
There’s little downside to doing that. This is not a performance-critical method. Let’s go with the more readable form.

Reply:
updated